CTS1.rst

```diff
@@ -0,0 +1,666 @@
+The contract name MUST be 100 bytes or less::
+
+  ContractName ::= (n: Byte²) (name: Byteⁿ)
+
+.. _CTS-AccountAddress:
+
+``AccountAddress``
+^^^^^^^^^^^^^^^^^^^
+
+An address of an account.
+
+It is serialized as 32 bytes::
+
+  AccountAddress ::= (address: Byte³²)
+
+.. _CTS-ContractAddress:
+
+``ContractAddress``
+^^^^^^^^^^^^^^^^^^^
+
+An address of a contract instance.
+It consists of an index and a subindex both unsigned 64 bit integers.
+
+It is serialized as: first 8 bytes for the index (``index``) followed by 8 bytes for the subindex (``subindex``) both little endian::
+
+  ContractAddress ::= (index: Byte⁸) (subindex: Byte⁸)
+
+.. _CTS-Address:
+
+``Address``
+^^^^^^^^^^^
+
+Is either an account address or a contract address.
+
+It is serialized as: First byte indicates whether it is an account address or a contract address.
+In case the first byte is 0 then ``AccountAddress`` (``address``) is followed.
+In case the first byte is 1 then ``ContractAddress`` (``address``) is followed::
+
+  Address ::= (0: Byte) (address: AccountAddress)
+            | (1: Byte) (address: ContractAddress)
+
+
+.. _CTS-Receiver:
+
+``Receiver``
+^^^^^^^^^^^^
+
+The receiving address of a transfer, which is either an account address or a contract address.
+In the case of a contract address, additional information such as the name of a receive function and some additional bytes to call on the receiving contract.
+
+It is serialized as: First byte indicates whether it is an account address or a contract address.
+In case the first byte is 0 then ``AccountAddress`` (``address``) is followed.
+In case the first byte is 1 then ``ContractAddress`` (``address``), bytes for :ref:`CTS-ReceiveHookName` (``hook``) is followed::
+
+    Receiver ::= (0: Byte) (address: AccountAddress)
+               | (1: Byte) (address: ContractAddress) (hook: ReceiveHookName)
+
+.. _CTS-AdditionalData:
+
+``AdditionalData``
+^^^^^^^^^^^^^^^^^^^
+
+Additional bytes to include in a transfer, can be used to add additional parameters for the transfer function call.
+
+It is serialized as: the first 2 bytes encodes the length (``n``) of the data, followed this many bytes for the data (``data``)::
+
+  AdditionalData ::= (n: Byte²) (data: Byteⁿ)
+
+.. note::
+
+  This type is passed in a parameter for smart contract function calls, be aware of the parameter size limit of 1024 bytes.
+
+.. _CTS-functions:
+
+Contract functions
+------------------
+
+A smart contract implementing CTS1 MUST export three functions :ref:`CTS-functions-transfer`, :ref:`CTS-functions-updateOperator` and :ref:`CTS-functions-balanceOf` according to the following description:
+
+.. _CTS-functions-transfer:
+
+``transfer``
+^^^^^^^^^^^^
+
+Executes a list of token transfers.
+A transfer is a token ID, an amount of this token ID to be transferred from one address to some other address.
```

Comment:
This sentence does not read very well. Maybe be clearer "A transfer is a token ID, an amount of token to transfer, and the from and to addresses."